db/migrate/20150311080458_associate_approvals_with_proposals.rb

```diff
@@ -0,0 +1,33 @@
+class AssociateApprovalsWithProposals < ActiveRecord::Migration
+  def change
+    add_reference :approvals, :proposal
+
+    reversible do |dir|
+      # http://stackoverflow.com/a/6257679/358804
+
+      dir.up do
+        execute <<-SQL
+          UPDATE approvals
+          SET proposal_id = proposals.id
+          FROM carts
+          -- get associated proposal through the cart
+          LEFT OUTER JOIN proposals ON carts.proposal_id = proposals.id
+          WHERE approvals.cart_id = carts.id;
```

Comment:
I spent waaay too long getting this to work – will definitely bug you for advice on the next one :wink: